cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -585,6 +599,28 @@ func (p *BicepProvider) getAppConfigsToPurge(
 	return configs, nil
 }
 
+func (p *BicepProvider) getAPIManagementsToPurge(
+	ctx context.Context,
+	groupedResources map[string][]azcli.AzCliResource,
+) ([]*azcli.AzCliAPIM, error) {
+	apims := []*azcli.AzCliAPIM{}
+
+	for resourceGroup, groupResources := range groupedResources {
```

Comment:
Might be worthwhile to add a comment here to note that soft-delete happens for all API Management resources, and that's why we don't have an additional level of filtering like we do on some of our other functions.